src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -164,6 +164,24 @@ public static FormulaValue FilterTable(EvalVisitor runner, SymbolContext symbolC
             return new InMemoryTableValue(irContext, rows);
         }
 
+        public static FormulaValue IndexTable(IRContext irContext, FormulaValue[] args)
+        {
+            var arg0 = (TableValue)args[0];
+            var arg1 = (NumberValue)args[1];
+
+            if ((int)arg1.Value < 1 || (int)arg1.Value >= arg0.Rows.Count())
```

Comment:
do this cast in one place, it's getting reused a few places.

Reply:
done